examples/discovery-search-app/cypress/integration/pagination/spec.ts

```diff
@@ -0,0 +1,212 @@
+    });
+
+    it('lists the correct number of total pages', () => {
+      cy.get('.bx--pagination__text').should('contain', '12');
+    });
+
+    it('the previous page button is disabled', () => {
+      cy.get('.bx--pagination__button--backward').should('be.disabled');
+    });
+
+    describe('and the next page arrow is clicked', () => {
+      beforeEach(() => {
+        cy.wait('@multiPageQuery');
+        cy.get('.bx--pagination__button--forward').click();
+      });
+
+      it('correctly requests the next page', () => {
+        cy.wait('@multiPageQuery')
+          .its('requestBody.offset')
+          .should('eq', 5);
+      });
+
+      describe('and the previous page arrow is clicked', () => {
+        beforeEach(() => {
+          cy.wait('@multiPageQuery');
+          cy.get('.bx--pagination__button--backward').click();
+        });
+
+        it('correctly requests the previous page', () => {
+          cy.wait('@multiPageQuery')
+            .its('requestBody.offset')
+            .should('eq', 0);
+        });
+      });
+    });
+
+    describe('and we use the page selector', () => {
+      beforeEach(() => {
+        cy.wait('@multiPageQuery');
+      });
+
+      it('we can navigate to each page of results', () => {
+        for (let i = 0; i < 12; i++) {
+          cy.get('.bx--pagination__right')
+            .find('.bx--select-input')
+            .select(`${i + 1}`);
+          cy.wait('@multiPageQuery')
+            .its('requestBody.offset')
+            .should('eq', i * 5);
+        }
+      });
+    });
+
+    describe('and we navigate to the last page of results', () => {
+      beforeEach(() => {
+        cy.wait('@multiPageQuery');
+        cy.get('.bx--pagination__right')
+          .find('.bx--select-input')
+          .select('12');
+      });
+
+      it('the next page button is disabled', () => {
+        cy.wait('@multiPageQuery');
+        cy.get('.bx--pagination__button--forward').should('be.disabled');
+      });
+
+      describe('and we increase the number of results per page to 50', () => {
+        beforeEach(() => {
+          cy.get('.bx--pagination__left')
+            .find('.bx--select-input')
+            .select('50');
+        });
+
+        it('returns to the first page, with the correct size', () => {
+          cy.wait('@multiPageQuery');
+          cy.wait('@multiPageQuery').then(xhr => {
+            expect(xhr.requestBody.count).to.eq(50);
+            expect(xhr.requestBody.offset).to.eq(0);
+          });
+          cy.get('.bx--pagination__right')
+            .find('.bx--select-input')
+            .should('have.value', '1');
+        });
+      });
+    });
+
+    describe('and items per page is set to 5', () => {
+      beforeEach(() => {
+        cy.get('.bx--pagination__left')
+          .find('.bx--select-input')
+          .select('5');
+      });
+
+      it('makes a request for 5 results', () => {
+        cy.wait('@multiPageQuery'); // not the request we're looking for
```

Comment:
curious why we have multiple requests - shouldn't it only be submitting one `POST /query`?

Reply:
Is one of these to retrieve the facets and one for the actual result set and if so are we unable to skip that initial query if we aren't requesting facets?

Reply:
I remember digging into the multiple query requests with @Steven-Gassert before the holidays. I think we narrowed it down to the facets, just as Gillin is describing. I'm not sure if there's a way to stop the app from sending the facets request, but we could probably make this a little bit cleaner by doing
```
cy.wait([@multiPageQuery, @multiPageQuery])
  .spread((_, multiPageQuery) => {
    //evaluate @multiPageQuery
  });
```

Reply:
seems like a bug if we have multiple requests being sent at the same time. i think we have a request being sent at page load, but we shouldn't be having multiple for pagination clicks

Reply:
I missed the fact there were multiple _per click_. Yeah that sounds bad.

Reply:
I cleaned up a bunch of the `cy.wait`s, and I think this is just the one page-load request, not a bunch of duplicates on subsequent clicks. Handling the `wait`s in the `beforeEach`s cut down on the sheer number, and adding one to the very beginning, right after `cy.visit('/')` removed the need for any of the other double-`wait`s. I'll push up an update that shows these changes shortly...